core/src/main/java/org/apache/iceberg/ManifestEntry.java

```diff
@@ -54,7 +57,12 @@ static Schema getSchema(StructType partitionType) {
   }
 
   static Schema wrapFileSchema(StructType fileType) {
-    return new Schema(STATUS, SNAPSHOT_ID, SEQUENCE_NUMBER, required(DATA_FILE_ID, "data_file", fileType));
+    // remove ManifestEntry fields from the file type when wrapping to avoid duplication
```

Comment:
I should note that this PR doesn't actually add the manifest entry fields to `DataFile`, it just sets up that change.